### PR TITLE
adding char * constructor for ParameterVarient 

### LINKED ISSUE #75: ParameterVariant constructor with hard coded string gets interpreted as bool
`rclcpp::parameter::ParameterVariant("bar", std::string("hello"))` is intepreted as a string correctly. 
`rclcpp::parameter::ParameterVariant("bar", "hello")` is interpreted as a bool with value true.

From testing in ros2/system_tests#14

### PR DESCRIPTION
**Pull request by tfoote**
fixes #75

### CODE REVIEW

File: rclcpp/include/rclcpp/parameter.hpp [hunk 1]
```diff
@@ -86,6 +86,12 @@ class ParameterVariant
     value_.string_value = string_value;
     value_.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
   }
+  explicit ParameterVariant(const std::string & name, const char * string_value)
+  : name_(name)
+  {
+    value_.string_value = std::string(string_value);
+    value_.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
+  }
```
Comment (dirk-thomas) [inline rclcpp/include/rclcpp/parameter.hpp]: Please call the constructor with the std::string argument instead of reimplementing the same body.

**Comment by wjwwood** [comment]
lgtm

**Comment by esteve** [comment]
+1 (including @dirk-thomas 's comment)

**Comment by tfoote** [comment]
switched to delegating constructor. That's a neat new c++11 feature.

**Comment by wjwwood** [comment]
+1 (again)

**Comment by dirk-thomas** [comment]
+1

**Pull request merged**